Makefile

```diff
@@ -3,7 +3,8 @@ help:
 	@echo "  test        run tests"
 
 deps:
-	pip install -r test_requirements.txt
+	pip install --user --requirement test_requirements.txt
+	pip install .
```

Comment:
hm, is this desirable? It will install the repo version of `twingly-search-api-python` in addition to the dependencies.

Reply:
Ah, you mean you do not want it installed on your local system, makes some sense.

I think it is desirable on CI, as it uses `setup.py` then.

Reply:
I think `make deps` should install just deps, nothing else. Even on Travis.

Reply:
But `pip install .` is also installing dependencies :)

Reply:
Hehe, a bit of a stretch. Anyway, seems like we agree that installing in local system is unexpected either way.

Reply:
I would like to use the same setup locally and on Travis to be able to catch errors like #40 and I want it to be the same so it doesn't go out of sync, but that seems impossible if we don't want to do `pip install .` locally. I will add something that can run whole testing locally with Docker then.

Reply:
The whole point of this PR was to do `pip install .` instead of `pip install -r requirements.txt`, so I don't see going back to `pip install -r requirements.txt` on Travis as an option. Do you have any other thoughts on it @walro?

Reply:
I see your point, meanwhile I would move this line back to `.travis.yml` as it will at least make travis pick up on the errors. 

Locally I would just add a line to the readme under development: `pip install -r requirements.txt` for now.

I find the risk for it to go out of date pretty low as locally you will find out soon enough that you are lacking required packages.

Reply:
I didn't see your comment before writine mine, but I think I answered it anyway:)

Reply:
Sure, let's do it like that.

I think I will remove the Makefile altogether, as we really can't share any stuff with it between CI and local environment. We also don't use Makefiles in our other projects, so our know-how with them is low (one idea I had was to branch on [`TRAVIS=true`](https://docs.travis-ci.com/user/environment-variables/#default-environment-variables) somewhere, but I don't know exactly what to write without reading some docs).

Reply:
OK, changed my mind, I will keep the Makefile, but make two separate "deps" targets (one for CI, one for local env).

Reply:
sounds good :9